### PR TITLE
Fix usage of -fsycl-device-code-split option

### PR DESCRIPTION
**Pull request by dmitriy-sobolev**
That fixes #691 without reverting changes. See #702 also.

### CODE REVIEW

**Comment by dmitriy-sobolev** [comment]
I am working on testing this draft as well as on adding `-fsycl-device-code-split` for all the tests using double types

**Comment by dmitriy-sobolev** [comment]
We have the following list of tests using double type:  [fp64.log](https://github.com/oneapi-src/oneDPL/files/10139994/fp64.log)
The list is large and hard to maintain, that's why it had better to distinguish a platform and then use `-fsycl-device-code-split` for all the tests.

**Comment by SergeyKopienko** [comment]
In this list https://github.com/oneapi-src/oneDPL/files/10139994/fp64.log we have only 88 tests from hundreds and hundreds of our tests. Are you sure in this idea ?

**Comment by dmitriy-sobolev** [comment]
> In this list https://github.com/oneapi-src/oneDPL/files/10139994/fp64.log we have only 88 tests from hundreds and hundreds of our tests. Are you sure in this idea?

I am. There are ~300 tests. We have to provide and maintain a separate config to cover those 88 tests. These can be 88 lines or a slightly smaller amount of regex expressions (~50 lines). Moreover, the fix going to be implemented in the near future: https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,15 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE $<$<CONFIG:Release>:-fno-strict-aliasing>)
     endif()
 
+    # Avoid speculative compilation of kernels which may not be supported by a device
+    # This options must be used with DPCPP backend, which requires -fsycl option
+    if (ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$")
```
Comment (SergeyKopienko) [inline test/CMakeLists.txt:45]: May be required to check ${split_kernels} at first ?

Comment (dmitriy-sobolev) [reply test/CMakeLists.txt:46]: It is irrelevant for the latest commit. I have removed `split_kernels` argument to enable `-fsycl-device-code-split` for all the tests.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -40,6 +40,25 @@ macro(onedpl_add_test test_source_file)
         target_compile_options(${_test_name} PRIVATE $<$<CONFIG:Release>:-fno-strict-aliasing>)
     endif()
 
+    # Avoid speculative compilation of kernels which may not be supported by a device
+    # This workaround is relevant until the following feature is implemented:
+    # https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md
+    if (ONEDPL_BACKEND MATCHES "^(dpcpp|dpcpp_only)$")
+        if (CMAKE_CXX_COMPILER_ID STREQUAL IntelLLVM)
+            # Do not use the workaround unintentionally with newer versions if the feature is implmented
+            if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 2023.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 2023.0)
```
Comment (akukanov) [inline test/CMakeLists.txt:49]: Would not `if (NOT ... VERSION_GREATER ...)` be better for now? We can later change to VERSION_LESS once the "good" version is known.

**Comment by dmitriy-sobolev** [comment]
Agreed with @SergeyKopienko that is is better to pass this option externally, e.g. when calling `cmake` command because of having information about specific hardware. Closing that PR.

**Pull request closed without merging**